### ISSUE TITLE
About display unavailability (and any about the project ?? )

### ISSUE THREAD

**Issue by Rohan5534** [joeycastillo/The-Open-Book#77]
Hey joey 👋 

Really nice project. Its unreal the amount of work you put in doing this , so kudos for that 👏 

I was wanting to make one for myself (abridged edition).
However the display on aliexpress page is out of stock. When I checked their own website (gooddisplay website), they don't have an option to ship display to my country (in fact they can ship to only a handful of countries. 

Is there some other replacement display for that display ??

Just another thought - The abridged edition is great if you want to get started. But I think it would be great if there was some other version that was more powerful (smth with ESP32 maybe). With features like WiFi , a bigger display (slightly bigger, not like 9-10", optionally backlit), using a rechargable battery.

What are your thoughts about this ?? Are you planning to make something similar or is it not in the books for few months ??
I understand making something like this involves insane amount of work (especially if u r going to put your time and money into and get probably no financial benefit because everything is open source). So if you need help with this new version about anything , I am open to help. I am no expert , but I have a fair knowledge about embedded stuff and electronics, so ill help as much as i can. 

Hope you have a great day 👋

**Comment by Niehztog** [comment]
I believe joey has recently done a new version of The Open Book with an ESP32. See his latest articles on the hackaday project page, for example https://hackaday.io/project/168761-the-open-book/log/221370-the-quest-for-a-sub-100-diy-e-book-reader
This gitrepo also contains the design files here https://github.com/joeycastillo/The-Open-Book/tree/main/OSO-BOOK-B1

**Comment by joeycastillo** [comment]
Yep the ESP32-S3 version is in the works! I wouldn’t recommend building the version in the repo just yet though; there’s at least one error (no pull-up on the EN line) and I haven’t fully tested everything yet. 

No idea what to do about the displays. I will need to reach out to Good Display; they had an update to this display pending (faster refresh, different driver) but I’m not sure if they pulled this display in advance of putting that one up, or if they’re just temporarily out of stock.

**Comment by d1nuc0m** [comment]
@joeycastillo I also contacted Good Display, they answered that the GDEW042T2 is nearly End Of Life and suggested to use instead a [GDEY042T81](https://www.good-display.com/product/386.html)  (that uses a different IC, SSD1683). Maybe you could contact them and explain the project requirements for a more suitable replacement

**Comment by d1nuc0m** [comment]
> but on the plus side, they both seem to use the same SSD1683 chip, which might make it the same effort for either one.

[In a issue](https://github.com/martinberlin/cale-idf/issues/90) a profile that should be related to Good Display commented that GDEQ042T81 and GDEY042T81 are "full compatible". By the way, as you're working on a ESP32 version, might a "[Epaper component driver for the ESP-IDF framework](https://github.com/martinberlin/cale-idf)" be interesting?

**Comment by Rohan5534** [comment]
Ok , sorry I was kinda busy last few days so couldnt see the replies.

But I am happy to know that you are already working on ESP32 version. Hope we get the working version soon. 

Would it be possible to make the display a little bit bigger than 4.2 inches ? Like this [5.83 inch display ](https://www.good-display.com/product/444.html) yes the full refresh is 3 seconds, but it says that it supports fast refresh of one second and partial refresh of 0.3 seconds, so maybe a full refresh once in a while wont hurt (waiting 3s for Each page turn is definately not smth that anyone would like, but if its 3s once say every 50 page turns, then its understandable ig ?)  ?? I am not too sure how it works, so maybe this display is worse ? But just wanted to see if it was possible.

Anyway , thanks for the replies everyone , hope yall have a great day

**Comment by chloe-everhart** [comment]
i'd love to be able to buy all the parts as a kit from Oddly Specific Objects.  Even better if I could buy a completely made Open Book, as low vision issues make it (the act of making an open book) so I'd need help to make it (the open book).

**Comment by Thanitorial** [comment]
hey all

appologies for how forward this is going to sound by asking...

just out of curiosity did a version of the software that will support the gdey042t81 get compiled at all?

asking as I have got a semi completed board that will not "post" signs of life (due to the fact I am running the gdey and not the gdew) and was hoping that the firmware was in some state of testing for the pico...

thanks in advance

**Comment by joeycastillo** [comment]
I have one assembled with the GDEY display but at the time I built it there wasn’t any existing driver code for it. I meant to write a driver but didn’t have time. 

I’m curious if other libraries like GxEPD have added support for this display; if so it might not be that difficult to get one of their demo sketches running on it. 

Also: there might be a couple of component swaps required to get the GDEY version working. In particular their spec shows a higher value inductor, a pulldown on one signal line and (maybe?) a different value for the current sense resistor. All the footprints to support this are present on the board; however, I had not published the new values since I have not had a chance to test a working version of the board with this display yet.

**Comment by Thanitorial** [comment]
Thanks for the rapid response joe

i did do a bit of surfing and found a driver [here](https://github.com/ZinggJM/GxEPD2). It is the GxEPD2 driver. 

I will give it a stab and see what I can get going but as you stated in the subsequent paragraph I may not have any luck as I am attempting to drive the screen with the existing display adapter board using the listed components(just arrived today).

Assuming there is no hardware adjustments to the PCB would it be possible you could either link to the spec or post here the new values if at all possible? If not I will try to find it but had to ask first as you seem to already have the info.

totally understand the time thing...

again thanks for the super rapid response.

**Comment by martinberlin** [comment]
> Also: there might be a couple of component swaps required to get the GDEY version working. In particular their spec shows a higher value inductor, a pulldown on one signal line and (maybe?) a different value for the current sense resistor

Might be but at least in my tests it’s 100% hardware compatible and you can just swap one for the other.

**Comment by Thanitorial** [comment]
do you have images of it working on the current production board? as i am running into a missing miso pin at gpio pin12 which is being used as the lock button pin. i have managed to get the screen to "work" somewhat by assigning the pin that would be gpio pin12 to gpio pin 8. i read on the datasheet that it can be controlled by the cs dc reset and busy pins but havent had luck with that just yet...

before anyone asks how, all i have used so far is the gxepd2 lib and the gxepd2_example arduino sketch which accounts for pico arch.

should i open up a seperate issue about this?

**Comment by Thanitorial** [comment]
Hey sorry for how long this has taken but I have got a waveshare example using 3 wire spi working with fantastic results on the gdey screen using [this](https://files.waveshare.com/upload/2/27/Pico_ePaper_Code.zip) example from their webpage [here](https://www.waveshare.com/wiki/Pico-ePaper-4.2-B).
going to try to break this down some more to attempt integration into something usable with Libros C architecture.

EDIT: Looking into Libros it seems like a library rewrite / better implementation of the 3 wire spi needs to be done (not saying it needs to be done just what is possibly broken). gonna try to break it open and see what is stopping it from sending anything over the spi line as I can get the display working on the example C script from waveshare and it works quite nicely (found great success in not defining a MISO pin which i am still looking into how it works for better understanding on attempting a DIY integration of 3 wire spi but cant seem to implement this without finding how deep the current method of SPI being used goes and a possible dual use of 4 wire and 3 wire spi as it is required for the memory and sd card read and writing)... @joeycastillo am I safe to assume that the library for the screen is an inhouse adaption of the adafruit libs and the gxepd libs?

**Comment by d1nuc0m** [comment]
Hi @joeycastillo any news on using a different display?
